src/BaGet/Program.cs

```diff
@@ -38,29 +40,27 @@ public static async Task Main(string[] args)
             {
                 var host = CreateWebHostBuilder(args).Build();
 
+                // Todo: Don't know how to get the options without rebuilding the host!
+                var baGetOptions = host.Services.GetRequiredService<IOptions<BaGetOptions>>();
+                host = CreateWebHostBuilder(args).UseUrls(baGetOptions.Value.Port).Build();
```

Comment:
WebHost is the legacy host, the generic host is the preferred host. You should be able to use this instead: https://docs.microsoft.com/en-us/aspnet/core/fundamentals/host/generic-host?view=aspnetcore-3.1#urls

```cs
Host.CreateDefaultBuilder(args)
    .ConfigureWebHostDefaults(webBuilder =>
    {
        // configure kestrel...

        if (has urls config)
        {
            web.UseUrls(urls config);
        }

        web.UseStartup<Startup>();
    });
```